src/Carrier/Factory/CarrierFactory.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\Pdk\Carrier\Factory;
+
+use MyParcelNL\Pdk\Base\Config;
+use MyParcelNL\Pdk\Base\Factory\PdkFactory;
+use MyParcelNL\Pdk\Carrier\Model\Carrier;
+use MyParcelNL\Pdk\Shipment\Collection\DefaultLogger;
+use MyParcelNL\Pdk\Tests\Bootstrap\MockConfig;
+
+class CarrierFactory
+{
+    public const KEY_CARRIER_ID    = 'id';
+    public const KEY_CARRIER_NAME  = 'name';
+    public const KEY_CONTRACT_ID   = 'contractId';
+    public const TYPE_NAME         = 'type';
+    public const TYPE_VALUE_CUSTOM = 'custom';
+    public const TYPE_VALUE_MAIN   = 'main';
+
+    private static $config = [];
+
+    /**
+     * @param  int|string $carrier
+     * @param  array|null $alternateConfig
+     *
+     * @return \MyParcelNL\Pdk\Carrier\Model\Carrier
+     * @throws \Exception
+     */
+    public static function create($carrier, array $alternateConfig = null): Carrier
+    {
+        self::$config = $alternateConfig ?? Config::get('carriers');
+
+        $followOrdered = [
```

Comment:
dit kan een constante worden, en geef dit dan ook meteen een naam waardoor we snappen wat het doet